packages/apps/bitbucket/src/services/bitbucket.ts

```diff
@@ -67,7 +68,17 @@ export default class BitBucketService extends TransactionBaseService {
     let documents: IndexableDocument[] = [];
 
     try {
-      const workspaces = await this.fetchWorkspaces(config);
+      let workspaces = []
+      if(auth.auth_strategy === AuthStrategy.API_TOKEN_STRATEGY){
+        if(!auth?.metadata?.workspace){
+          throw new Error("No workspace found in metadata")
```

Comment:
while throwing the error make sure that backend does not crash like , the error must be caught in some outer try-catch block